Sources_v3/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -263,21 +263,19 @@ extension DatabaseContainer {
     /// Iterates over all items and if the DTO conforms to `EphemeralValueContainers` calls `resetEphemeralValues()` on
     /// every object.
     func resetEphemeralValues() {
-        write({ session in
-            let session = session as! NSManagedObjectContext
-            
-            try self.managedObjectModel.entities.forEach { entityDescription in
-                guard let entityName = entityDescription.name else { return }
-                let fetchRequest: NSFetchRequest<NSFetchRequestResult> = NSFetchRequest(entityName: entityName)
-                let entities = try session.fetch(fetchRequest) as? [EphemeralValuesContainer]
-                entities?.forEach { $0.resetEphemeralValues() }
-            }
-        }, completion: { error in
-            if let error = error {
-                log.error("Error resetting ephemeral values: \(error)")
-            } else {
+        writableContext.performAndWait {
```

Comment:
I checked call stack for this method, and unless I missed something, it will be called on main thread.
So for with increasing storage, it will take more and more time, meaning locking main thread for longer and longer

Reply:
`performAndWait` documentation:
> Synchronously performs a given block on the *context’s queue*.
`writableContext` is a background context, so it shouldn't be executed on main thread, interesting that you've seen such behavior

Reply:
Yes, main point is: `Synchronously performs`

Reply:
Ooh you mean that point.. As I wrote in the discussion part of the description, I didn't see considerable delay, but that's just anectodal.. So what you've raised is correct, and I need suggestions on resolving this completely